contrib/Findsodium.cmake

```diff
@@ -32,7 +32,6 @@ if (CMAKE_C_COMPILER_ID STREQUAL "GNU"
 endif()
 
 # static library option
-option(sodium_USE_STATIC_LIBS "enable to statically link against sodium")
```

Comment:
instead of removing this, I would very much prefer something like this:
```cmake
set(sodium_USE_STATIC_LIBS OFF CACHE BOOL "enable to statically link against sodium")
```
which can be overridden by just setting up the cached variable before issuing a call to `find_package(sodium)`. See the [docs](https://cmake.org/cmake/help/v3.10/command/set.html#set-cache-entry) for reference

Reply:
From the calling side, I want to be able to do this:

```
set(sodium_USE_STATIC_LIBS ON)
find_package(sodium REQUIRED)
```

This doesn't work if you "declare" the variable inside the find module using `option()` or with `set( .... CACHE ...)`.  Just looking around at the various Find* modules provided with CMake, you'll see that most of them *don't* try to declare the variable -- probably because it prevents the simple usage I've outlined above. Yes, if we declare it a CACHE variable in the module, I could override it in my CMakeLists, but I would have to do:

```
set(sodium_USE_STATIC_LIBS ON CACHE BOOL "why are you making me do this?")
```

So, for that reason, I advise we simply check the variable and not try to declare it as a cache/option - again, that is what most standard Find* modules do.

Reply:
The rationale behind the switch was that the precompiled binaries for windows always provide shared and static versions and it really is a matter of preference whether or not to use static or shared libraries, so from my point of view it made sense to provide a visible option.

However, I do understand that there are projects for which this agility isn't wanted/necessary and I want to support those too, so I think guarding the option with an if not defined clause is the most sensible approach.

Lastly I recommend squashing all those changes into one commit in order to reduce the noise created by these small changes.

Reply:
ok - I think the defined check around option works for me. Squashed and pushed - let me know if it looks right to you.